src/task_queue/nudges/user_event_nudge.py

```diff
@@ -90,42 +94,78 @@ def get_email_lists(users):
             emails.append(user.email)
 
     return emails
-          
+
 
 def update_last_notification_dates(email):
     new_date = str(datetime.date.today())
-    user =  UserProfile.objects.filter(email=email).first()
+    user = UserProfile.objects.filter(email=email).first()
     user_notification_dates = user.notification_dates if user.notification_dates else {}
 
-    notification_dates = {**user_notification_dates,"user_event_nudge":new_date}
-    UserProfile.objects.filter(email=email).update(**{"notification_dates":notification_dates })
+    notification_dates = {**user_notification_dates, "user_event_nudge": new_date}
+    UserProfile.objects.filter(email=email).update(**{"notification_dates": notification_dates})
+
 
+def is_event_eligible(event, community_id, task=None):
+    now = timezone.now()
+    settings = event.nudge_settings.filter(communities__id=community_id).first()
+
+    if settings.never:
+        return False
+
+    freq = task.frequency if task else None
+    last_last_run = None
+
+    freq_to_delta = {
+        "EVERY_WEEK": relativedelta(weeks=1),
+        "bi-weekly": relativedelta(weeks=2),
+        "EVERY_MONTH": relativedelta(months=1),
+        "EVERY_DAY": relativedelta(days=1)
+    }
+
+    if freq:
+        last_last_run = now - freq_to_delta.get(freq, relativedelta(days=0))
+
+    if settings.when_first_posted and event.published_at and last_last_run < event.published_at <= now:
+        return True
+    elif settings.within_30_days and event.start_date_and_time - now <= timezone.timedelta(days=30):
+        return True
```

Comment:
The intention of "within_30_days" was that a user should be notified once about it when the event is about 30 days away.  As I read this, the user will be nudged every week when the event is less than 30 days away.

Reply:
Yes, the current implementation will send the notification multiple times within 30 days. This issue will be addressed alongside the new direction outlined in @frimpongopoku's [Figma design](https://www.figma.com/file/broFyNtjZQINatcfJxShqA/Community-Profile-Rearrangment?type=design&node-id=359-2&mode=design).